mmgen/models/architectures/biggan/modules.py

```diff
@@ -0,0 +1,416 @@
+            self.in_channels // 2,
+            kernel_size=1,
+            padding=0,
+            bias=False,
+            act_cfg=None,
+            with_spectral_norm=with_spectral_norm,
+            spectral_norm_cfg=dict(eps=sn_eps))
+        self.o = SNConvModule(
+            self.in_channels // 2,
+            self.in_channels,
+            kernel_size=1,
+            padding=0,
+            bias=False,
+            act_cfg=None,
+            with_spectral_norm=with_spectral_norm,
+            spectral_norm_cfg=dict(eps=sn_eps))
+        # Learnable gain parameter
+        self.gamma = Parameter(torch.tensor(0.), requires_grad=True)
+
+    def forward(self, x):
+        # TODO:
+        """[summary]
+
+        Args:
+            x ([type]): [description]
+
+        Returns:
+            [type]: [description]
+        """
+        # Apply convs
+        theta = self.theta(x)
+        phi = F.max_pool2d(self.phi(x), [2, 2])
+        g = F.max_pool2d(self.g(x), [2, 2])
+        # Perform reshapes
+        theta = theta.view(-1, self.in_channels // 8, x.shape[2] * x.shape[3])
+        phi = phi.view(-1, self.in_channels // 8, x.shape[2] * x.shape[3] // 4)
+        g = g.view(-1, self.in_channels // 2, x.shape[2] * x.shape[3] // 4)
+        # Matmul and softmax to get attention maps
+        beta = F.softmax(torch.bmm(theta.transpose(1, 2), phi), -1)
+        # Attention map times g path
+        o = self.o(
+            torch.bmm(g, beta.transpose(1, 2)).view(-1, self.in_channels // 2,
+                                                    x.shape[2], x.shape[3]))
+        return self.gamma * o + x
+
+
+@MODULES.register_module()
+class BigGANDiscResBlock(nn.Module):
+    # TODO:
+    """[summary]
+
+    Args:
+        in_channels ([type]): [description]
+        out_channels ([type]): [description]
+        conv_cfg ([type], optional): [description]. Defaults to dict(type=
+            'Conv2d').
+        shortcut_cfg ([type], optional): [description]. Defaults to dict(
+                type='Conv2d').
+        act_cfg ([type], optional): [description]. Defaults to dict(type=
+            'ReLU', inplace=False).
+        sn_eps ([type], optional): [description]. Defaults to 1e-6.
+        with_downsample (bool, optional): [description]. Defaults to True.
+        with_spectral_norm (bool, optional): [description]. Defaults to
+            True.
+        head_block (bool, optional): [description]. Defaults to False.
+    """
+
+    def __init__(
+        self,
+        in_channels,
+        out_channels,
+        conv_cfg=dict(type='Conv2d'),
+        shortcut_cfg=dict(type='Conv2d'),
+        act_cfg=dict(type='ReLU', inplace=False),
+        sn_eps=1e-6,
+        with_downsample=True,
+        with_spectral_norm=True,
+        head_block=False,
+    ):
+        super().__init__()
+        self.activation = build_activation_layer(act_cfg)
+        self.with_downsample = with_downsample
+        self.head_block = head_block
```

Comment:
`is_head_block`? What's the meaning of `head _block`. In the forward function, I guess that you use this variable to control the order of modules?